From d1f73856a3d46fc90671110f5066038625dfc1b9 Mon Sep 17 00:00:00 2001
From: unknown <sinclairs@.(none)>
Date: Thu, 22 May 2008 17:06:01 -0700
Subject: [PATCH] MingW does not need the gai_strerrorA() function defined in server.c.

---
 src/server.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/server.c b/src/server.c
index df3c898..209865d 100644
--- a/src/server.c
+++ b/src/server.c
@@ -68,6 +68,7 @@ static void queue_data(lo_server s, lo_timetag ts, void *data, size_t len);
 
 
 #ifdef WIN32
+#if 0
 // Copied from the Win32 SDK 
 
 // WARNING: The gai_strerror inline functions below use static buffers,
@@ -93,7 +94,7 @@ char *WSAAPI gai_strerrorA(int ecode)
                               NULL);
     return buff;
 }
-
+#endif
 
 static int stateWSock = -1;
 
-- 
1.5.5.1015.g9d258

