diff -rui liblo-0.23-msvc7-orig/lo/lo.h liblo-0.23-changed/lo/lo.h
--- liblo-0.23-msvc7-orig/lo/lo.h	2007-04-06 18:56:57.718750000 -0400
+++ liblo-0.23-changed/lo/lo.h	2007-04-06 16:58:21.406250000 -0400
@@ -237,6 +237,18 @@
  */
 EXPORT lo_server lo_server_thread_get_server(lo_server_thread st);
 
+/**
+ * \brief Set the server's dispatch_method callback function.
+ */
+void lo_server_thread_set_dispatch_callback(lo_server_thread st,
+                                            lo_dispatch_method_callback callback);
+
+/**
+ * \brief Set the server's handler callback function.
+ */
+void lo_server_thread_set_handler_callback(lo_server_thread st,
+                                           lo_handler_callback callback);
+
 /** \brief Return true if there are scheduled events (eg. from bundles) waiting
  * to be dispatched by the thread */
 EXPORT int lo_server_thread_events_pending(lo_server_thread st);
diff -rui liblo-0.23-msvc7-orig/lo/lo_lowlevel.h liblo-0.23-changed/lo/lo_lowlevel.h
--- liblo-0.23-msvc7-orig/lo/lo_lowlevel.h	2007-04-06 18:56:57.734375000 -0400
+++ liblo-0.23-changed/lo/lo_lowlevel.h	2007-04-06 16:48:53.546875000 -0400
@@ -484,6 +484,21 @@
 double lo_server_next_event_delay(lo_server s);
 
 /**
+ * \brief Set the server's dispatch_method callback function.
+ */
+void lo_server_set_dispatch_callback(lo_server s, lo_dispatch_method_callback callback);
+
+/**
+ * \brief Set the server's handler callback function.
+ */
+void lo_server_set_handler_callback(lo_server s, lo_handler_callback callback);
+
+/**
+ * \brief Dispatch the given method to registered handler functions.
+ */
+void lo_server_dispatch_method(lo_server s, const char *data, size_t len);
+
+/**
  * \brief Return the protocol portion of an OSC URL, eg. udp, tcp.
  *
  * This library uses OSC URLs of the form: osc.prot://hostname:port/path if the
diff -rui liblo-0.23-msvc7-orig/lo/lo_types.h liblo-0.23-changed/lo/lo_types.h
--- liblo-0.23-msvc7-orig/lo/lo_types.h	2005-04-25 17:46:47.000000000 -0400
+++ liblo-0.23-changed/lo/lo_types.h	2007-04-06 16:52:57.531250000 -0400
@@ -123,6 +123,21 @@
 				 lo_arg **argv, int argc, lo_message msg,
 				 void *user_data);
 
+/**
+ * \brief A callback function used to intercept method calls before they
+ * are dispatched to a handler.  This function must make a call to
+ * lo_server_dispatch_method() with the arguments provided.
+ */
+typedef void (*lo_dispatch_method_callback)(void* s, const char *data, size_t len);
+
+/**
+ * \brief A callback function used to intercept handler calls.
+ */
+typedef int (*lo_handler_callback)(lo_method_handler handler,
+                 const char *path, const char *types,
+				 lo_arg **argv, int argc, lo_message msg,
+				 void *user_data);
+
 #ifdef __cplusplus
 }
 #endif
diff -rui liblo-0.23-msvc7-orig/src/lo_types_internal.h liblo-0.23-changed/src/lo_types_internal.h
--- liblo-0.23-msvc7-orig/src/lo_types_internal.h	2005-08-11 05:07:29.000000000 -0400
+++ liblo-0.23-changed/src/lo_types_internal.h	2007-04-06 16:52:57.468750000 -0400
@@ -53,6 +53,12 @@
 	struct _lo_method *next;
 } *lo_method;
 
+typedef void (*lo_dispatch_method_callback)(void* s, const char *data, size_t len);
+typedef int  (*lo_handler_callback)(lo_method_handler handler,
+                 const char *path, const char *types,
+				 lo_arg **argv, int argc, lo_message msg,
+				 void *user_data);
+
 typedef struct _lo_server {
 	int	                 socket;
 	struct addrinfo         *ai;
@@ -65,6 +71,8 @@
 	void		        *queued;
 	struct sockaddr_storage  addr;
 	socklen_t 	         addr_len;
+    lo_dispatch_method_callback dispatch_callback;
+    lo_handler_callback handler_callback;
 } *lo_server;
 
 typedef struct _lo_server_thread {
diff -rui liblo-0.23-msvc7-orig/src/server.c liblo-0.23-changed/src/server.c
--- liblo-0.23-msvc7-orig/src/server.c	2007-04-06 18:56:57.796875000 -0400
+++ liblo-0.23-changed/src/server.c	2007-04-06 16:54:27.765625000 -0400
@@ -62,8 +62,7 @@
 
 static int lo_can_coerce_spec(const char *a, const char *b);
 static int lo_can_coerce(char a, char b);
-static void dispatch_method(lo_server s, const char *path, char *types,
-			    void *data);
+static void dispatch_method(lo_server s, const char *data, size_t len);
 static int dispatch_queued(lo_server s);
 static void queue_data(lo_server s, lo_timetag ts, void *data, size_t len);
 
@@ -165,6 +164,8 @@
     s->path = NULL;
     s->queued = NULL;
     s->socket = -1;
+    s->dispatch_callback = NULL;
+    s->handler_callback = NULL;
 
     memset(&hints, 0, sizeof(hints));
 
@@ -574,7 +575,7 @@
 	    if ((ts.sec == 0 && ts.frac == 1) ||
 				lo_timetag_diff(ts, now) <= 0.0) {
 		types = pos + lo_strsize(pos);
-		dispatch_method(s, pos, types + 1, types + lo_strsize(types));
+		dispatch_method(s, pos, len);
 	    } else {
 		queue_data(s, ts, pos, len);
 	    }
@@ -590,7 +591,7 @@
 	return -1;
     }
 
-    dispatch_method(s, path, types+1, data);
+    dispatch_method(s, data, size);
 
     free(data);
 
@@ -616,9 +617,20 @@
     return 100.0;
 }
 
-static void dispatch_method(lo_server s, const char *path, char *types,
-			    void *data)
+static void dispatch_method(lo_server s, const char *data, size_t len)
 {
+    if (s->dispatch_callback)
+        s->dispatch_callback(s, data, len);
+    else
+        lo_server_dispatch_method(s, data, len);
+}
+
+void lo_server_dispatch_method(lo_server s, const char *_data, size_t len)
+{
+	const char *path = _data;
+	const char *types = path + lo_strsize(path) + 1;
+	void *data = types + lo_strsize(types);
+
     int argc = strlen(types);
     lo_arg **argv = NULL;
     lo_method it;
@@ -718,7 +730,11 @@
 		*/
 		pptr = path;
 		if (it->path) pptr = it->path;
-		ret = it->handler(pptr, types, argv, argc, msg,
+        if (s->handler_callback)
+            ret = s->handler_callback(it->handler, pptr, types,
+                      argv, argc, msg, it->user_data);
+        else
+    		ret = it->handler(pptr, types, argv, argc, msg,
 				      it->user_data);
 
 	    } else if (lo_can_coerce_spec(types, it->typespec)) {
@@ -753,7 +769,11 @@
 		*/
 		pptr = path;
 		if (it->path) pptr = it->path;
-		ret = it->handler(pptr, it->typespec, argv, argc, msg,
+        if (s->handler_callback)
+            ret = s->handler_callback(it->handler, pptr, types,
+                      argv, argc, msg, it->user_data);
+        else
+    		ret = it->handler(pptr, types, argv, argc, msg,
 				      it->user_data);
 		free(argv);
 		free(data_co);
@@ -898,7 +918,8 @@
 	path = ((queued_msg_list *)s->queued)->data;
 	types = path + lo_strsize(path) + 1;
 	data = types + lo_strsize(types);
-	dispatch_method(s, path, types, data);
+	dispatch_method(s, ((queued_msg_list *)s->queued)->data,
+                       ((queued_msg_list *)s->queued)->len);
 
 	free(((queued_msg_list *)s->queued)->data);
 	free((queued_msg_list *)s->queued);
@@ -1062,6 +1083,17 @@
     }
 }
 
+void lo_server_set_dispatch_callback(lo_server s,
+                                     lo_dispatch_method_callback callback)
+{
+    s->dispatch_callback = callback;
+}
+
+void lo_server_set_handler_callback(lo_server s, lo_handler_callback callback)
+{
+    s->handler_callback = callback;
+}
+
 static int lo_can_coerce_spec(const char *a, const char *b)
 {
     unsigned int i;
diff -rui liblo-0.23-msvc7-orig/src/server_thread.c liblo-0.23-changed/src/server_thread.c
--- liblo-0.23-msvc7-orig/src/server_thread.c	2007-04-06 18:56:57.812500000 -0400
+++ liblo-0.23-changed/src/server_thread.c	2007-04-06 17:07:52.656250000 -0400
@@ -129,6 +129,18 @@
     return lo_server_events_pending(st->s);
 }
 
+void lo_server_thread_set_dispatch_callback(lo_server_thread st,
+                                            lo_dispatch_method_callback callback)
+{
+    lo_server_set_dispatch_callback(st->s, callback);
+}
+
+void lo_server_thread_set_handler_callback(lo_server_thread st,
+                                           lo_handler_callback callback)
+{
+    lo_server_set_handler_callback(st->s, callback);
+}
+
 static void thread_func(void *data)
 {
     lo_server_thread st = (lo_server_thread)data;
